packages/headless-styles/sandbox/plugins/css-to-ts/convert-rules/convertToJS.js

```diff
@@ -52,12 +52,39 @@ const convertRules = (rules, res = {}) => {
   return result
 }
 
+function applyCompositions(obj) {
+  const result = {}
+  for (const [className, value] of Object.entries(obj)) {
+    if (typeof value !== 'object') {
+      result[className] = value
+      continue
+    }
+    const { composes, ...overrides } = value
+    result[className] = {}
+    if (composes) {
+      if (composes in obj) {
+        Object.assign(result[className], obj[composes])
+      } else if (composes.includes(' from ')) {
+        // Ignore here, will be appended in later step
+        Object.assign(result[className], value)
+      } else {
+        composes.split(' ').forEach((otherKey) => {
```

Comment:
I would split on 1 or more whitespace characters, rather than a single space character, just in case.

```suggestion
        composes.split(/\s+/).forEach((otherKey) => {
```